### PR TITLE
Feature/mx 1660 ldap search endpoint

### PR DESCRIPTION
**Pull request by vyvytranngoc**
# PR Context
- we want to be a able to manually pull items from auxiliary extractors to link them to other items in the mex-editor
- the editor therefore needs an api endpoint for each auxiliary extractor to search for items from an aux x-systems and return them in as extracted items

# Added
- provides api endpoint for Active Directory (aka LDAP)

### CODE REVIEW

File: tests/assets/raw-data/organigram/organizational_units.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
+[
```
Comment (cutoffthetop) [inline tests/assets/raw-data/organigram/organizational_units.json:1]: where are these two files needed?

File: mex/backend/auxiliary/organigram.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from mex.common.models import ExtractedOrganizationalUnit, ExtractedPrimarySource
```
Comment (cutoffthetop) [inline mex/backend/auxiliary/organigram.py:1]: this entire file is unused

File: mex/backend/auxiliary/ldap.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            paginated_persons,
+            extracted_primary_source_ldap(),
+            extracted_organizational_unit(),
+        )
+    )
+    return AuxiliarySearch(items=extracted_persons, total=count_persons)
+
+
+@cache
+def extracted_primary_source_ldap() -> ExtractedPrimarySource:
```
Comment (cutoffthetop) [inline mex/backend/auxiliary/ldap.py:61]: please add a unittest for this function

File: mex/backend/auxiliary/ldap.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    Args:
+        q: The name of the person to be searched.
+        offset: The starting index for pagination
+        limit: The maximum number of results to return
+
+    Returns:
+        paginated list of ExtractedPerson and the total count of persons found.
+    """
+    count_persons = get_count_of_found_persons_by_name(q)
+    found_persons = list(get_persons_by_name(q))
```
Comment (cutoffthetop) [inline mex/backend/auxiliary/ldap.py:48]: this will only search by `surname`

Comment (vyvytranngoc) [reply mex/backend/auxiliary/ldap.py:47]: actually no. in` ldap/extract.py` I allow to apply the search string on each field: `def get_persons_by_name(
    surname: str = "*",
    given_name: str = "*",
    **filters: str`

Comment (cutoffthetop) [reply mex/backend/auxiliary/ldap.py:49]: but how could that work?

the signature of `get_persons_by_name` is:

```python
def get_persons_by_name(
    surname: str = "*",
    given_name: str = "*",
    **filters: str,
) -> Generator[LDAPPerson, None, None]:
    ...
```

if you call it like `get_persons_by_name(q)`, you will only fill the `surname` argument, but never the `given_name`

**Review by cutoffthetop — CHANGES_REQUESTED** [review]
i have a few further questions

**Comment by vyvytranngoc** [comment]
> i have a few further questions
Running into this error with trivy. I'm not sure how to handle it:
grafik

**Comment by cutoffthetop** [comment]
> Running into this error with trivy. I'm not sure how to handle it

@vyvytranngoc sadly that's a known issue on trivy's side https://github.com/aquasecurity/trivy/discussions/7668

for now, you can try restarting the cve check by clicking "re-run job", usually i get it to pass after 1-2 retries.

mid-term, if they don't fix it on their end, we need to come up with our own solution..

**Comment by vyvytranngoc** [comment]
> > Running into this error with trivy. I'm not sure how to handle it
> 
> @vyvytranngoc sadly that's a known issue on trivy's side [aquasecurity/trivy#7668](https://github.com/aquasecurity/trivy/discussions/7668)
> 
> for now, you can try restarting the cve check by clicking "re-run job", usually i get it to pass after 1-2 retries.
> 
> mid-term, if they don't fix it on their end, we need to come up with our own solution..

After re-running it passed all the checks now :)

File: tests/auxiliary/test_ldap.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+from mex.common.types import (  # noqa: F811
+    ExtractedPrimarySourceIdentifier,
+    MergedPrimarySourceIdentifier,
+    Text,
+    TextLanguage,
+)
+from mex.common.types.identifier import (
+    ExtractedOrganizationalUnitIdentifier,
+    MergedOrganizationalUnitIdentifier,
+)
```
[the remark this answers is not in the document]
Comment (cutoffthetop) [reply tests/auxiliary/test_ldap.py:23]: you were importing `Text` from `typing` and then overwriting that with an import from `mex.common.types`. you have to be more careful when adding `# noqa` to lines and try to understand why you would want to ignore a specific rule there

File: mex/backend/auxiliary/ldap.py (hunk 4 above)
Comment (vyvytranngoc) [reply mex/backend/auxiliary/ldap.py:49]: Ah I see what you mean. Hmm, how would like to handle it? Adding an additional param for the givenname or handling the query syntax like `q="family-name:{famName]+AND+given-name:{givName}"` ?

Comment (cutoffthetop) [reply mex/backend/auxiliary/ldap.py:49]: there is also a property `displayName` on ldap persons, which will be the full name of a person including all family and given names. it should be safe to search through those. luckily the get_persons_by_name also accepts arbitrary filter arguments, so you can use that directly here:

```python
found_persons = list(get_persons_by_name(displayName=q))
```

File: tests/auxiliary/test_ldap.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (vyvytranngoc) [reply tests/auxiliary/test_ldap.py]: So I've found the problem. I've changed mex common code on the mex backend repo. Of course It did'nt appear as a diff :)
New changes pushed -> https://github.com/robert-koch-institut/mex-common/pull/350

[the remark this answers is not in the document]
Comment (cutoffthetop) [reply tests/auxiliary/test_ldap.py]: aah, okay! the mex-common PR looks good, aside from a small formality ✅

**Review by cutoffthetop — APPROVED** [review]
nice ✨ thanks for fixing the displayName thing! this is ready to merge now.

**Pull request merged**